### PR TITLE
movine: init at 0.11.0

### PR DESCRIPTION
**Pull request by netcrns**
###### Motivation for this change
We want [`movine`](https://github.com/byronwasti/movine) packaged upstream.

###### Things done
Packaged [`movine`](https://github.com/byronwasti/movine) to be merged upstream.

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/tools/database/movine/default.nix
Comment (legendofmiracles) [inline pkgs/development/tools/database/movine/default.nix:26]:
```suggestion
    description = ''A migration manager written in Rust that attempts to be smart yet minimal'';
```
This is usually a oneliner

**Review by legendofmiracles — CHANGES_REQUESTED** [review]

File: pkgs/development/tools/database/movine/default.nix
Comment (netcrns) [reply pkgs/development/tools/database/movine/default.nix:26]: Should I just commit this suggestion with the GitHub-preset message "Update pkgs/development/tools/database/movine/default.nix", or what's the convention here?

**Comment by legendofmiracles** [comment]
Trying to compile this locally doesn't work, you have to add another commit `maintainers: add <real name/username>` to add yourself as a maintainer in `maintainers/maintainers-list.nix`

File: pkgs/development/tools/database/movine/default.nix
Comment (legendofmiracles) [reply pkgs/development/tools/database/movine/default.nix:26]: make a new commit, but then [squash](https://gitbetter.substack.com/p/how-to-squash-git-commits) it into the first

**Comment by netcrns** [comment]
> Trying to compile this locally doesn't work because you aren't in the maintainer list, you have to add another commit `maintainers: add <real name/username>` to add yourself as a maintainer in `maintainers/maintainers-list.nix`

I've added myself to the maintainers-list [here](https://github.com/NixOS/nixpkgs/pull/120010), in an open PR.
What's the etiquette in this situation? Just do it again here?

**Comment by legendofmiracles** [comment]
I'd wait for the other one to be merged, as there could be merge conflicts otherwise, and having the same commit twice is dumb.

**Comment by netcrns** [comment]
> I'd wait for the other one to be merged, as there could be merge-conflicts otherwise, and having the same commit twice is dumb.

Seeing as this PR should presumably be merged before the aforementioned one (we're waiting for sozu v0.12.0 to be released), I'll remove the commit from that and move it here. Would that be alright?

**Comment by legendofmiracles** [comment]
If you do it correctly (maybe with git-cherry-pick, and remove it from the other PR) I don't see why not.

**Review by legendofmiracles — APPROVED** [review]

**Comment by legendofmiracles** [comment]
Whoops misclick 😅

File: pkgs/development/tools/database/movine/default.nix
Comment (legendofmiracles) [reply pkgs/development/tools/database/movine/default.nix]: Resolve the conversation, since you fixed this

**Comment by legendofmiracles** [comment]
Result of `nixpkgs-review pr 120754` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>movine</li>
  </ul>
</details>

**Review by legendofmiracles — APPROVED** [review]
If the bot doesn't anymore complain about it not working on darwin, then it LGTM

File: pkgs/development/tools/database/movine/default.nix
Comment (OPNA2608) [inline pkgs/development/tools/database/movine/default.nix:16]:
```suggestion
  nativeBuildInputs = [ pkg-config ];
```

File: pkgs/development/tools/database/movine/default.nix
Comment (OPNA2608) [inline pkgs/development/tools/database/movine/default.nix:19]:
```suggestion
  buildInputs = [ postgresql sqlite ] ++ (
    if !stdenv.isDarwin then [ openssl ] else [ Security ]
  );
```

File: pkgs/development/tools/database/movine/default.nix
Comment (OPNA2608) [inline pkgs/development/tools/database/movine/default.nix:1]:
```suggestion
{ rustPlatform, fetchFromGitHub, lib, stdenv, pkg-config, postgresql, sqlite, openssl, Security }:
```

File: pkgs/top-level/all-packages.nix
Comment (OPNA2608) [inline pkgs/top-level/all-packages.nix:16502]:
```suggestion
  movine = callPackage ../development/tools/database/movine {
    postgresql = postgresql_12;
    inherit (darwin.apple_sdk.frameworks) Security;
  };
```

**Review by OPNA2608 — CHANGES_REQUESTED** [review]
- Dont use `pkgs.<…>` in derivations, always explicitly specify what packages you need at the top.
- Darwin-specific frameworks are usually used by `inherit`ing them on the `callPackage` level & again specifying only their package names in the derivation.
- I think it's usually better to use the default package name `postgresql` in the derivation and set it to a specific version `postgresql_12` on the `callPackage` level. You can disagree and leave it as-is if you want, but be aware that overriding the `postgresql` version will look abit more awkward.
  `movine.override { postgresql_12 = postgresql_13; }`
  That aside, does `postgresql_12` need to be pinned to this version? If `postgresql` (`postgresql_11` currently) would work for this package as well then you should use that instead and `override` it locally to whatever version you'd prefer. If it wouldn't, maybe a comment to explain why that's the case would help.

Please also reorder the commits. First should be the commit that adds you to the maintainers list, second should be the package init.

File: pkgs/top-level/all-packages.nix
Comment (netcrns) [reply pkgs/top-level/all-packages.nix:16502]: I was not able to build this with unstable nix any more, so I discarded those changes.
Is this strictly necessary? I will have to look into why it fails on unstable, in that case.

Comment (OPNA2608) [reply pkgs/top-level/all-packages.nix:16502]: Checking out your PR & applying the suggested changes evaluates fine for me.

grafik

If you want to `callPackage` the default.nix yourself, then you also need to replicate what the `callPackage` in all-packages.nix does.

`nix-build -E 'with import <nixpkgs> {}; callPackage ./default.nix { postgresql = postgresql_12; inherit (darwin.apple_sdk.frameworks) Security; }'`

Comment (netcrns) [reply pkgs/top-level/all-packages.nix:16502]: ah, yeah, good point lmao
works fine now, cheers

**Comment by OPNA2608** [comment]
Result of `nixpkgs-review pr 120754` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>movine</li>
  </ul>
</details>

I'll try to run a `nixpkgs-review` on my Darwin machine this evening. If that passes too, LGTM.

File: pkgs/development/tools/database/movine/default.nix
Comment (OPNA2608) [inline pkgs/development/tools/database/movine/default.nix:10]:
```suggestion
, libiconv
}:
```

File: pkgs/development/tools/database/movine/default.nix
Comment (OPNA2608) [inline pkgs/development/tools/database/movine/default.nix:27]:
```suggestion
    if !stdenv.isDarwin then [ openssl ] else [ Security libiconv ]
```

**Review by OPNA2608 — CHANGES_REQUESTED** [review]
Fails to build on Darwin.

> ```
> ld: library not found for -liconv
> ```

**Review by OPNA2608 — APPROVED** [review]
Result of `nixpkgs-review pr 120754` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>movine</li>
  </ul>
</details>

Builds & `movine help` works, LGTM.

**Pull request merged**
